packages/firebase_core/ios/Classes/FirebaseCorePlugin.m

```diff
@@ -58,20 +71,20 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
       options.storageBucket = optionsDictionary[@"storageBucket"];
     if (![optionsDictionary[@"deepLinkURLScheme"] isEqual:[NSNull null]])
       options.deepLinkURLScheme = optionsDictionary[@"deepLinkURLScheme"];
-    if (![name isEqual:[NSNull null]]) {
-      [FIRApp configureWithName:name options:options];
-    } else {
-      [FIRApp configureWithOptions:options];
-    }
+    [FIRApp configureWithName:name options:options];
     result(nil);
   } else if ([@"FirebaseApp#allApps" isEqualToString:call.method]) {
     NSDictionary<NSString *, FIRApp *> *allApps = [FIRApp allApps];
     NSMutableArray *appsList = [NSMutableArray array];
     for (NSString *name in allApps) {
       FIRApp *app = allApps[name];
-      [appsList addObject:@{@"name" : app.name, @"options" : app.options.dictionary}];
+      [appsList addObject:app.flutterDictionary];
     }
     result(appsList.count > 0 ? appsList : nil);
+  } else if ([@"FirebaseApp#appNamed" isEqualToString:call.method]) {
+    NSString *name = call.arguments;
+    FIRApp *app = [FIRApp appNamed:name];
+    result(app.flutterDictionary);
```

Comment:
Crash if there is no app with that name?

Reply:
This is Objective-C so it just returns `nil`

Reply:
Ah, ok. I thought that was only the case with `[message passing]`. But perhaps property access is just msg passing under the hood?

Reply:
Yes